### PR TITLE
GitHub user via code

### PR DESCRIPTION
**Pull request by JoshMock**
Putting optional env vars in the Dockerfile `CMD` was breaking the Helm chart deployment. Moving the logic into the code.

My first time committing production Perl!

### CODE REVIEW

**Comment by JoshMock** [comment]
Hmm. CI error says `INFO:build_docs:Can't locate URI.pm`. @nik9000 maybe you can school me on Perl module imports in the morning?

File: build_docs.pl [hunk 1]
```diff
@@ -555,9 +556,14 @@ sub init_target_repo {
 #===================================
     my ( $repos_dir, $temp_dir, $reference_dir ) = @_;
 
+    my $git_uri = URI->new($Opts->{target_repo});
+    if ( $ENV{GITHUB_USER} ){
+        $git_uri->userinfo( $ENV{GITHUB_USER} . ":" . $ENV{GITHUB_PASS} );
+    }
+
```
Comment (nik9000) [inline build_docs.pl]: I think we should probably only try and do this if it *look* like an http uri. The regular docs build uses git-over-ssh and probably should get this decoration. Same for tests that reference the filesystem.

Comment (JoshMock) [reply build_docs.pl]: Added a check in cccabfdc3ab2f26f094309ce54701a1f1da2ad2b to ensure it's using https before attaching creds to the URI.

Also, as of e3063d6126ab1faab103c778f145f6408e767371, it falls back to the originally-passed argument if the string's scheme is not https.

**Comment by JoshMock** [comment]
Woo hoo, all checks have passed!

I just republished Docker image version 12 now, overwriting it since it's not used by anything yet. @nik9000 this is ready for review again!

**Comment by JoshMock** [comment]
thanks for the helpful review @nik9000!

**Pull request merged**